ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java

```diff
@@ -40,35 +40,61 @@
 
   static final short VERSION_0 = 0;
   static final short VERSION_1 = 1;
+  static final short VERSION_2 = 2;
 
   private static final int VERSION_SIZE = 2;
   private static final int TYPE_SIZE = 2;
   private static final int SESSION_ID_LENGTH_SIZE = 4;
+  private static final int INCARNATION_ID_LENGTH_SIZE = 4;
 
   private static final byte[] ZERO_LENGTH_ARRAY = new byte[0];
   private static final int UNINITIALIZED_OFFSET = -1;
 
-  private final Type type;
-  private final Offset offset;
-  private final StoreKey storeKey;
-  private final UUID sessionId;
-  private long bytesRead;
+  private final Type type; // refers to the type of the token
+  private final Offset offset; // refers to the offset in the log. Could be either of Journal or Index based token
+  private final StoreKey storeKey; // refers to the store key incase of Index based token
+  private final byte inclusive; // incase of journal based token, represents if the blob at the offset(of the token)
+  // is inclusive or not
+  private final UUID sessionId; // refers to the sessionId of the store. On every restart a new sessionId is created
+  private final UUID incarnationId; // refers to the incarnationId of the store. On every re-creation of the store,
+  // a new incarnationId is created
+  private long bytesRead; // refers to the bytes read so far (from the beginning of the log)
 
+  /**
+   * Uninitialized token. Refers to the starting of the log.
+   */
   StoreFindToken() {
-    this(Type.Uninitialized, null, null, null);
+    this(Type.Uninitialized, null, null, null, null, true);
   }
 
-  StoreFindToken(StoreKey key, Offset indexSegmentStartOffset, UUID sessionId) {
-    this(Type.IndexBased, indexSegmentStartOffset, key, sessionId);
+  /**
+   * Index based token. Refers to an index start offset and a store key expected to belong to that index segment.
```

Comment:
"expected" might not be the right word here. "Store key that belongs to that index segment" is more appropriate - any other state is a bug.